### PR TITLE
feat(database): Support total Prisma migration

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change, and why?
Support the scenario where a database migration has gone wrong, wrong enough that we've deleted the `_prisma_migrations` table.

There are now two new commands:
  1. `npm -w repocop run migrate:code:from-start`
  2. `npm -w repocop run migrate:prod:from-start`

When used, these scripts will first perform a [baseline migration](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#baseline-your-production-environment) to avoid data loss. The remaining migrations will then be applied as normal.

To enable this, the existing migrations have been edited to allow them to be infinitely replayed. This just means `IF EXISTS` has been added.

Note, updating previous migration scripts changes the file's checksum. As the checksum is a column in the `_prisma_migrations` table, we'd need to perform `npm -w repocop run migrate:prod:from-start` on PROD upon merge.

## How has it been verified?
I've successfully run `npm -w repocop run migrate:code:from-start` on CODE multiple times.

### CODE REVIEW

File: packages/common/prisma/migrations/20230907105726_create_table_repocop_github_repository_rules/migration.sql [hunk 1]
```diff
@@ -1,3 +1,5 @@
+DROP TABLE IF EXISTS repocop_github_repository_rules;
```
Comment (AshCorr) [inline packages/common/prisma/migrations/20230907105726_create_table_repocop_github_repository_rules/migration.sql:1]: Why is this table in particular dropped?

**Review by AshCorr — APPROVED** [review]
I feel like "total prisma migration" sounds way too cool for what its actually doing

File: packages/common/prisma/migrations/20230928080813_guardian_non_p_and_e_github_teams/migration.sql [hunk 2]
```diff
@@ -1,3 +1,5 @@
+DROP TABLE IF EXISTS guardian_non_p_and_e_github_teams;
```
Comment (AshCorr) [inline packages/common/prisma/migrations/20230928080813_guardian_non_p_and_e_github_teams/migration.sql:1]: and this table too? is there a chance someone might manually add a value to this table which will be lost if we do a total migration?

**Comment by akash1810** [comment]
Confirming that I've applied this to PROD this morning using:

```sh
npm -w repocop run migrate:prod:from-start
```

**Pull request merged**